### PR TITLE
fix: ensure badge always applies inline-style when attributes exist

### PR DESCRIPTION
**Pull request by chrisdholt**
# Description
The badge component had odd behavior when passing inline styles and there was a scenario you could end up in where nothing would be applied if fill was not passed. This change creates a function to ensure attributes are handled correctly here.

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [x] **Bug fix**: A change that fixes an issue, link to the issue above.
- [ ] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

**Adding or modifying component(s) in `@microsoft/fast-components` checklist**

- [ ] I have added a new component
- [x] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

## Process & policy checklist

- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### CODE REVIEW

**Review by scomea — APPROVED** [review]

File: packages/web-components/fast-foundation/src/badge/badge.ts [hunk 1]
```diff
@@ -36,4 +36,21 @@ export class Badge extends FASTElement {
      */
     @attr({ mode: "boolean" })
     public circular: boolean;
+
+    public generateBadgeStyle = () => {
+        const fill: string = `background-color: var(--badge-fill-${this.fill});`;
+        const color: string = `color: var(--badge-color-${this.color});`;
+
+        if (!this.fill && !this.color) {
```
Comment (nicholasrice) [inline packages/web-components/fast-foundation/src/badge/badge.ts:44]: You should move this early-exit condition to the top so we don't allocate variables if they aren't needed.

Comment (chrisdholt) [reply packages/web-components/fast-foundation/src/badge/badge.ts:41]: moved 👍 - thanks!

File: packages/web-components/fast-foundation/src/badge/badge.template.ts [hunk 2]
```diff
@@ -7,14 +7,7 @@ import { Badge } from "./badge";
...
     <template class="${x => (x.circular ? "circular" : "")}">
-        <div
-            class="control"
-            part="control"
-            style="${x =>
-                x.fill || x.color
-                    ? `background-color: var(--badge-fill-${x.fill}); color: var(--badge-color-${x.color})`
-                    : void 0}"
-        >
+        <div class="control" part="control" style="${x => x.generateBadgeStyle()}">
```
Comment (nicholasrice) [inline packages/web-components/fast-foundation/src/badge/badge.template.ts:10]: Does this update if fill or color change?

Comment (chrisdholt) [reply packages/web-components/fast-foundation/src/badge/badge.template.ts:10]: I tested by changing color in the DOM and it updated with the expected colors.

**Review by EisenbergEffect — APPROVED** [review]

**Review by nicholasrice — APPROVED** [review]

**Pull request merged**